nestedtensor/nested/creation.py

```diff
@@ -30,3 +19,12 @@ def nested_tensor(data, dtype=None, device=None, requires_grad=False, pin_memory
     if pin_memory:
         result = result.pin_memory()
     return result
+
+
+def as_nested_tensor(data, dtype=None, device=None):
+    # TODO: Needs tests to check failure cases
```

Comment:
Just do it(c)

Reply:
Follow-up PR

Reply:
Because there is potentially a lot of edge cases